specs/0118-notification-center.md

```diff
@@ -0,0 +1,119 @@
+- Header: `Message type`
+  - Selects the type of message to be sent
+  - Checkboxes: `Email`, `Notification` (multiselect, default: both boxes empty)
+  - Validation: Error if no checkbox selected - `You must select at least one message type.`
+- Header: `Message editor`
+  - Creates a subject and body for emails. Subject line will be required for in-app messages; in app-messages have the subject line combined with the body message as the first line of the message separated by a newline.
+  - Field: `Subject` (text input)
+  - Field: `Body` (text input with formatting: markdown or html)
+  - Button: `Preview` - shows a rendered version of the text in `Body` for checking formatting, similar to how the preview for GitHub comments work
+  - Validation: 
+    - Error if `Subject` field empty: `Your message must include a subject.`
+    - Error if `Body` field empty: `Your message must include a body.`
+- Header: `Recipients`
+  - Select who will receive a message based on a combination of the below filters.
+  - Checkbox: `Filter by User Role` (default empty) - enables User Role as a filter.
+    - Checkboxes: `Manager`, `Translator`, `Contributor` (multiselect, default: all boxes empty)
+    - Validation:
+      - Error if `Filter by User Role` enabled but no roles selected: `You must select at least one user role.`
+  - Checkbox: `Filter by Locale` (default empty) - enables Locale as a filter.
+    - Use existing 2-panel widget already used for selecting locales in Pontoon
+    - Choose locales by moving them from `Available` to `Chosen`(all locales available)
+    - Include a `Move All` option
+    - Validation:
+      - Error if `Filter by Locale` enabled but no locales selected: `You must select at least one locale.`
+  - Checkbox: `Projects`
+    - Use existing 2-panel widget already used for selecting locales in Pontoon
+    - Choose projects by moving them from `Available` to `Chosen` (all projects available)
+    - Include a `Move All` option
+  - Subheader `Filter by Activity`
+    - Button: `Add activity filter`
+      - Creates a new section of the form labeled `Activity filter` and an associated button `Remove filter`.
+      - Creates a dropdown list `Activity type` associated with this filter
+      - `Add activity filter` button persists at the bottom of the form and will still be visible after being clicked, allowing multiple filters to be created
+    - Dropdown list: `Activity type`
+      - Options: `Select activity type`, `Submitted translations`, `Performed reviews`, `Last login`
+      - Select one
+      - Default value: `Select activity type`
+      - Form dynamically displays the appropriate filter inputs below based on the selection in the dropdown. `Select activity type` does not display any filter fields.
+      - Validation:
+        - Multiple activity filters can used at the same time
+        - Error if more than one filter of the same type used: `You cannot use multiple of the same activity type.`
```

Comment:
Can we prevent this by populating the dropdown dynamically? If you've already selected "Last login", that disappears from the list.